apps/condo/domains/user/schema/_oidc.test.js

```diff
@@ -120,39 +118,50 @@ test('oidc', async () => {
+        redirect_uris: [uri], // using uri as redirect_uri to show the ID Token contents
         response_types: ['code id_token'],
         token_endpoint_auth_method: 'client_secret_basic',
     })
 
     const nonce = generators.nonce()
-    const scope = 'openid'
-    const url = new URL(serverSideOidcClient.authorizationUrl({ scope, nonce }))
+    const url = new URL(serverSideOidcClient.authorizationUrl({ nonce }))
     const oidcAuthUrl = `${c.serverUrl}${url.pathname}${url.search}`
 
     // 2) client side ( open oidcAuthUrl )
 
     const res1 = await request(oidcAuthUrl, c.getCookie())
-    expect(res1.status).toBe(200)
+    expect(res1.status).toBe(303)
+    expect(res1.headers.location.startsWith('/oidc/interaction/')).toBeTruthy()
     expectCookieKeys(res1.cookie, ['keystone.sid', '_interaction', '_interaction.sig', '_interaction_resume', '_interaction_resume.sig'])
 
-    const redirectTo = `${c.serverUrl}${'/oidc' + res1.data.redirectTo.split('/oidc')[1]}`
+    const res2 = await request(`${c.serverUrl}${res1.headers.location}`, res1.cookie)
+    expect(res2.status).toBe(303)
+    expect(res2.headers.location.startsWith(`${c.serverUrl}/oidc/auth/`)).toBeTruthy()
+    expectCookieKeys(res2.cookie, ['keystone.sid', '_interaction', '_interaction.sig', '_interaction_resume', '_interaction_resume.sig'])
 
-    const res2 = await request(redirectTo, res1.cookie) // redirect to jwt.io
-    expect(res2.status).toBe(200)
-    expectCookieKeys(res2.cookie, ['keystone.sid', '_interaction', '_interaction.sig', '_interaction_resume', '_interaction_resume.sig', '_session', '_session.sig', '_session.legacy', '_session.legacy.sig'])
+    const res3 = await request(res2.headers.location, res2.cookie)
+    expect(res3.status).toBe(303)
+    expect(res3.headers.location.startsWith(`${uri}#code`)).toBeTruthy()
+    expectCookieKeys(res3.cookie, ['keystone.sid', '_interaction', '_interaction.sig', '_interaction_resume', '_interaction_resume.sig', '_session', '_session.sig', '_session.legacy', '_session.legacy.sig'])
 
     // 3) callback to server side ( callback with code to oidc app site; server get the access and cann use it )
 
-    const params = serverSideOidcClient.callbackParams(res2.url.replace('https://jwt.io/#', 'https://jwt.io/?'))
-    const tokenSet = await serverSideOidcClient.callback('https://jwt.io/', { code: params.code }, { nonce })
+    const params = serverSideOidcClient.callbackParams(res3.headers.location.replace(`${uri}#`, `${uri}?`))
+    const tokenSet = await serverSideOidcClient.callback(uri, { code: params.code }, { nonce })
     expect(tokenSet.access_token).toHaveLength(43) // important to be 43!
     expect(tokenSet.id_token.length > 10).toBeTruthy()
 
     // 4) check requests by accessToken to /oidc/userinfo
 
     const userinfo = await serverSideOidcClient.userinfo(tokenSet.access_token)
-    expect(userinfo).toEqual({ 'sub': c.user.id })
+    expect(userinfo).toEqual({
+        'sub': c.user.id,
```

Comment:
What 'sub' means?

Reply:
Subject Identifier. It's a part of OIDC spec: https://openid.net/specs/openid-connect-core-1_0.html#UserInfoResponse